zap/src/main/java/org/zaproxy/zap/model/Tech.java

```diff
@@ -232,7 +232,7 @@ public int compareTo(Tech o) {
      */
     public static void add(Tech tech) {
         Assert.isNonEmpty(tech);
-        if (tech.getParent() != null) {
+        if (tech.getParent() == null) {
```

Comment:
Its going to add null techs ?

Reply:
No it isn't that one is a check if we have a top-level/root tech element (no parent = new root)

Reply:
I fixed the null check, and Tech with parent null, means top level tech like Tech.Db.